tests/snapshot.tests.js

```diff
@@ -0,0 +1,110 @@
+import { createDocFromSnapshot, Doc, snapshot, YMap } from '../src/internals'
```

Comment:
Thanks for creating tests! These are very valuable!

We should also generate tests that contain multiple users that create changes that depend on each other. E.g. user1 inserts text and syncs with user2. User2 inserts text (the inserted `Item` depends on the content from user1).

Reply:
@dmonad thanks, do you mean we need to create a document with these dependent changes and then restore it to a specific state from the past? By generating tests, do you mean that they should be automatically generated somehow(in tests I saw `applyRandomTests` probably used for tests generation)?

Reply:
>  thanks, do you mean we need to create a document with these dependent changes and then restore it to a specific state from the past? 

Exactly!

> By generating tests, do you mean that they should be automatically generated somehow(in tests I saw applyRandomTests probably used for tests generation)?

You don't have to at first. But it does make sense to do it before we release this feature. A lot of issues can be found with randomized testing. 

Maybe we can adapt `applyRandomTests` to create a snapshot at some point. At the end of the test we restore a document and check if the old values can be recovered.